StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/SavedPaymentOptionsViewController.swift

```diff
@@ -508,8 +518,15 @@ extension SavedPaymentOptionsViewController: UICollectionViewDataSource, UIColle
             stpAssertionFailure()
             return UICollectionViewCell()
         }
-        cell.setViewModel(viewModel, cbcEligible: cbcEligible, allowsPaymentMethodRemoval: self.configuration.allowsRemovalOfPaymentMethods)
+        cell.setViewModel(viewModel, cbcEligible: cbcEligible, allowsPaymentMethodRemoval: self.configuration.allowsRemovalOfPaymentMethods, allowsSetAsDefaultPM: configuration.allowsSetAsDefaultPM, showDefaultPMBadge: hasDefault)
         cell.delegate = self
+        let isDefaultPM: Bool = {
+            guard self.hasDefault,
+                  let savedPMId = viewModel.savedPaymentMethod?.stripeId,
+                  let defaultPMId = self.elementsSession.customer?.defaultPaymentMethod else { return false }
+            return savedPMId == defaultPMId
+        }()
```

Comment:
This logic feel awkward because self.hasDefault is calling `self.elementsSession.customer?.defaultPaymentMethod`, which is also being called here. Seems like there is a way to simplify this logic -- maybe we can introduce a function which takes the stripeId and returns a boolean to determine if it is the default payment method?